frontend/pkg/frontend/external_auth.go

```diff
@@ -540,6 +543,120 @@ func (f *Frontend) updateExternalAuthInCosmos(ctx context.Context, writer http.R
+	if database.IsResponseError(err, http.StatusNotFound) {
+		// For resource not found errors on deletion, ARM requires
+		writer.WriteHeader(http.StatusNoContent)
+		return nil
+	}
+	if err != nil {
+		return utils.TrackError(err)
+	}
+
+	transaction := f.dbClient.NewTransaction(externalAuth.ID.SubscriptionID)
+	if err := f.addDeleteExternalAuthToTransaction(ctx, writer, request, transaction, externalAuth); err != nil {
+		return utils.TrackError(err)
+	}
+	_, err = transaction.Execute(ctx, nil)
+	if err != nil {
+		return utils.TrackError(err)
+	}
+
+	writer.WriteHeader(http.StatusAccepted)
+	return nil
+}
+
+func (f *Frontend) addDeleteExternalAuthToTransaction(ctx context.Context, writer http.ResponseWriter, request *http.Request, transaction database.DBTransaction, externalAuth *api.HCPOpenShiftClusterExternalAuth) error {
+	logger := utils.LoggerFromContext(ctx)
+
+	correlationData, err := CorrelationDataFromContext(ctx)
+	if err != nil {
+		return utils.TrackError(err)
+	}
+
+	if externalAuth.Properties.ProvisioningState == arm.ProvisioningStateDeleting {
+		return nil
+	}
+	if err := checkForProvisioningStateConflict(ctx, f.dbClient, database.OperationRequestDelete, externalAuth.ID, externalAuth.Properties.ProvisioningState); err != nil {
+		return utils.TrackError(err)
+	}
+
+	err = f.clusterServiceClient.DeleteExternalAuth(ctx, externalAuth.ServiceProviderProperties.ClusterServiceID)
+	var ocmError *ocmerrors.Error
+	if errors.As(err, &ocmError) && ocmError.Status() == http.StatusNotFound {
+		// StatusNotFound means we have stale data in Cosmos DB.
+		// This can happen in test environments if a user bypasses
+		// the RP to delete a resource (e.g. "ocm delete"). It can
+		// also happen if an asynchronous deletion operation fails.
+		// we will fall through and cancel all operations and go through as normal a deletion flow as we can to avoid
+		// leaking data related to the resource, like controller status.
+		logger.Info("clusterService nodepool missing, trying to clean up", "err", err)
+	} else if err != nil {
+		return utils.TrackError(err)
+	}
+
+	// Cluster Service will take care of canceling any ongoing operations
+	// on the resource or child resources, but we need to do some database
+	// bookkeeping to reflect that.
+	err = f.CancelActiveOperations(ctx, transaction, &database.DBClientListActiveOperationDocsOptions{
+		ExternalID:             externalAuth.ID,
+		IncludeNestedResources: true,
+	})
+	if err != nil {
+		return utils.TrackError(err)
+	}
+
+	operationDoc := database.NewOperationDocument(
+		database.OperationRequestDelete,
+		externalAuth.ID,
+		externalAuth.ServiceProviderProperties.ClusterServiceID,
+		"",
+		"",
+		"",
+		correlationData)
+	if request != nil {
+		// these are optional because when this is triggered via the subscription deletion flow, there is no notification URI
+		// so these operations cannot be directly tracked.
```

Comment:
```suggestion
		// these are optional because when this is triggered via the subscription deletion flow, there is no
		// deletion request containing these headers so these operations cannot be directly tracked.
```